code/AssetLib/FBX/FBXDocument.h

```diff
@@ -80,6 +80,10 @@ class BlendShape;
 class Skin;
 class Cluster;
 
+#define new_LazyObject new (allocator.Allocate(sizeof(LazyObject))) LazyObject
+#define new_Connection new (allocator.Allocate(sizeof(Connection))) Connection
+#define delete_LazyObject(_p) (_p)->~LazyObject()
+#define delete_Connection(_p) (_p)->~Connection()
```

Comment:
Object construction and destruction should be delegated to the function related to the allocator. Generally it would be better not to use macross like that overall. 
```cpp
// #include <new>

public:
    template<class T, class... Args>
    T* ConstructObject(Args&&... args)
    {
        return ::new(this->Allocate(sizeof(T))) T(std::forward<Args>(args)...);
    }

    template<class T>
    void DestroyObject(T* obj) noexcept
    {
        obj->~T();
    }
```